frontend/src/components/form/BookingsSearch.tsx

```diff
@@ -36,20 +40,37 @@ export default function Search({
   getSearchResults: Function;
 }) {
   const [locations, setLocations] = useState<Location[]>([] as Location[]);
-  const [userLocation, setUserLocation] = useState<Location | null>(null);
+  const [userLocation, setUserLocation] = useState<Location | null | undefined>(
+    null
+  );
   useEffect(() => {
     async function fetchLocation() {
       const fetchedLocations: Location[] = await getLocations();
       setLocations(fetchedLocations);
       const fetchedUserLocation: Location = await getUserLocation();
-      setUserLocation(fetchedUserLocation);
+      if (fetchedUserLocation) {
+        setUserLocation(
+          fetchedLocations.find((l) => l.id === fetchedUserLocation?.id)
+        );
+        getSearchResults({
+          url: '/booking',
+          method: 'GET',
+          params: { locationId: fetchedUserLocation?.id },
+        });
+      } else {
+        setUserLocation(null);
+      }
     }
     fetchLocation();
   }, [setLocations, setUserLocation]);
+
+  const { addAlert } = useAlert();
+
   return (
     <Box>
       <Formik
-        initialValues={Initial}
+        initialValues={{ ...Initial, locationId: userLocation?.id }}
+        enableReinitialize
```

Comment:
not sure why this isn't on by default in the formik library

Reply:
ha, that's true.